multiversx_sdk_cli/projects/templates.py

```diff
@@ -1,254 +1,55 @@
-        project_contract_name = self._contract_name(self.project_name)
-
-        self._replace_in_files(
-            source_paths,
-            [
-                # Example: replace contract name "pub trait SimpleERC20" to "pub trait MyContract"
-                (f"pub trait {template_contract_name}", f"pub trait {project_contract_name}"),
-                # Example: replace "simple_erc20.wasm" to "my_token.wasm"
-                (f"{self.template_name}.wasm", f"{self.project_name}.wasm"),
-                # Example: replace "use simple_erc20::*" to "use my_token::*"
-                (f"use {template_name}::*", f"use {project_name}::*"),
-                # Example: replace "<simple_erc20::AbiProvider>()" to "<my_token::AbiProvider>()"
-                (f"<{template_name}::AbiProvider>()", f"<{project_name}::AbiProvider>()"),
-                # Example: replace "extern crate adder;" to "extern crate myadder;"
-                (f"extern crate {template_name};", f"extern crate {project_name};"),
-                # Example: replace "empty::ContractObj" to "foo_bar::ContractObj"
-                (f"{template_name}::ContractObj", f"{project_name}::ContractObj"),
-                (f"{template_name}::ContractBuilder", f"{project_name}::ContractBuilder"),
-                (f"{template_name}::contract_obj", f"{project_name}::contract_obj"),
-            ],
-            ignore_missing
-        )
-
-    def _patch_source_code_tests(self):
-        test_dir_path = self.directory / "tests"
-        if not test_dir_path.is_dir():
-            return
-
-        test_paths = utils.list_files(test_dir_path)
-        self._patch_source_code_files(test_paths, ignore_missing=False)
+        self.template = template
+        self.path = path
 
-    def _patch_scenarios_tests(self):
-        test_dir_path = self.directory / "scenarios"
-        if not test_dir_path.is_dir():
-            return
+    def get_contract_templates(self) -> str:
+        self._check_if_dependencies_installed()
+        args = self._prepare_args_to_list_templates()
+        templates = myprocess.run_process(args=args, dump_to_stdout=False)
+        return templates
 
-        test_paths = utils.list_files(test_dir_path, suffix=".json")
-        self._replace_in_files(
-            test_paths,
-            [
-                (f"{self.template_name}.wasm", f"{self.project_name}.wasm")
-            ],
-            ignore_missing=False
-        )
+    def create_from_template(self) -> None:
+        self._check_if_dependencies_installed()
+        args = self._prepare_args_to_create_new_contract_from_template()
+        myprocess.run_process(args)
 
-        for file in test_paths:
-            data = utils.read_json_file(file)
-            # Patch fields
-            data["name"] = data.get("name", "").replace(self.template_name, self.project_name)
-            utils.write_json_file(str(file), data)
+    def _check_if_dependencies_installed(self):
```

Comment:
Can be renamed to `_ensure_dependencies_installed()`, perhaps? Since it does not only perform a simple check, also installs rust if needed etc.

Reply:
renamed to `_ensure_dependencies_installed()`